feat: Security manager role HRZ-185 (#1752)
codacy/docs: docs/organizations/managing-security-and-risk.md

| Medium                  | Medium        | 90            |
| Minor                   | Low           | 120           |

## Access management<!-- TODO Rename this. We're not only managing access, are we? -->

> Organization owners can assign the Security Manager role to organization members, as well as revoke it. Security managers share some permissions with organization owners:<!-- TODO State here more clearly why this is useful as a feature -->

Reviewer: Maybe we could explain first what is the Security Manager role - what it can do, mention that it doesn't map any Git provider permissions - and only after say who can grant/revoke and how.
WDYT?

|Permission|Organization Owner|Security Manager|
|---|---|---|
|Access Security and Risk Management|Yes|Yes|
|Access originating items<!-- TODO reword all "originating items" instances for clarity --> (Codacy repositories, Jira issues)|Yes<!-- TODO admin may not have access here? Confirm -->|Inherits original permissions|
|Manage integrations|Yes|<!-- TODO confirm if security managers cans manage integrations -->|
|Assign and revoke the Security Manager role|Yes|No|

To assign the Security Manager role to a user:

> 1.  Access the **Security and risk management configuration** page.

Reviewer: It's not clear how to this configuration page... 🤔

Author: 👍 Updated with clearer instructions

| Medium                  | Medium        | 90            |
| Minor                   | Low           | 120           |

> ## Access management

Reviewer 2: I like how you're considering adding this new section directly under the Managing security and risk feature. It has the advantage of keeping things more self-contained, at least for now. :+1:

Author: Correct, and as you're saying...at least for now. We now have three PRs targeting the same branch (which is itself a PR...), some with quite extensive changes. I'm keeping things as isolated as possible, at least on these lesser updates.

Organization owners can assign the Security Manager role to organization members, as well as revoke it.

Security managers share some capabilities with organization owners:

|Capability|Organization Owner|Security Manager|
|---|---|---|
|Access Security and Risk Management|Yes|Yes|
|Manage integrations|Yes|?<!-- TODO confirm if security managers cans manage integrations -->|
|Assign and revoke the Security Manager role|Yes|No|

To assign the Security Manager role to a user:

1.  Access the **Security and Risk Management configuration** page.

> 1.  In the Permissions area, use the search field to find the user of interest.

Reviewer 2: A shorter way of saying "the user of interest":

```suggestion
1.  In the Permissions area, use the search field to find the relevant user.
```

| Medium                  | Medium        | 90            |
| Minor                   | Low           | 120           |

## Access management<!-- Rename this. We're not really only managing access, are we? -->

> Organization owners can assign the Security Manager role to organization members, as well as revoke it. Security managers share some permissions with organization owners:

Reviewer 2: In particular, we should state here more clearly why organization owners may want to assign the Security Manager role to other people in their teams (or put into other words, why this is a useful feature).

Still dependent on the answers we get to the questions in the comments, for now it seems that the main advantage is that Security Managers will be able to access the Security and Risk dashboard.

|Permission|Organization Owner|Security Manager|
|---|---|---|
|Access Security and Risk Management|Yes|Yes|

> |Access originating items (Codacy repositories, Jira issues)|Yes<!-- TODO admin may nothave access here? Confirm -->|Inherits original permissions|

Reviewer 2: Is it possible to clarify better what it means to access "originating items"? I'm thinking that it's navigating to the details of those items (or just viewing the item details), although I'm not sure how that will work with Jira items. :thinking:

Reviewer 2: I just came across [this other line](https://github.com/codacy/docs/pull/1722/files#diff-30a735ab496ba1650575435a2195b32444487dbb269bb4b685b2e1ad8c7ebe53R37) and I should note that the two references should be kept in sync. :+1:

Author: This should link to either a filtered view of the Codacy issue list or directly to the Jira issue. I was looking for something generic enough to cover "Codacy issue", "Jira issue", and whatever we may include in the future.
```suggestion
|Access originating items<!-- TODO reword all "originating items" instances for clarity --> (Codacy repositories, Jira issues)|Yes<!-- TODO admin may not have access here? Confirm -->|Inherits original permissions|
```

|Manage integrations|Yes|<!-- TODO confirm if security managers cans manage integrations -->|
|Assign and revoke the Security Manager role|Yes|No|

To assign the Security Manager role to a user:

> 1.  Access the **Security and Risk Management configuration** page.

Reviewer 2: The page actually uses sentence case for the title (which is a good thing! :star:). :nerd:

```suggestion
1.  Access the **Security and risk management configuration** page.
```

> 1.  In the Permissions area, use the search field to find the relevant user.

Reviewer 2: We should use bold consistently when writing the names of UI elements:

```suggestion
1.  In the **Permissions** area, use the search field to find the relevant user.
```

Author: ```suggestion
1.  In the **Permissions** area, use the search field to find the relevant user and click the user's name.
```

    ![Security and Risk Management access management](images/security-risk-management-access-management.png)

> 1.  Click the relevant row.<!-- TODO reword this -->

Reviewer 2: It's up to you, but I'd probably merge this last (rather obvious) step with the previous one. Something like "[...] find and click the relevant user" perhaps?

To revoke the Security Manager role from a user:

1.  Access the **Security and Risk Management configuration** page.

> 1.  In the Permissions area, scroll the list to find the relevant user.

Reviewer 2: ```suggestion
1.  In the **Permissions** area, scroll the list to find the relevant user.
```

1.  Click the **Revoke role icon**<!-- TODO icon image here? --> to the right of the user's name and confirm.

> <!-- TODO screenshot highlighting the Permissions area -->

Author: TODO Remove this

Reviewer 2: We should be able to get away with a single screenshot for both operations. :+1:

| **Medium**                     | 90                 | Medium                             | Medium                           |
| **Low**                        | 120                | Minor                              | Low and other                    |

## Managing access to Security and Risk Management

To better track and address security vulnerabilities, organization admins can extend access to Security and Risk Management by assigning the Security Manager role to organization members.

The Security Manager role is independent of the Git provider role of an organization member and provides only the permissions necessary to monitor security vulnerabilities, following the principle of least privilege:

| Permission                                                  | Organization Admin | Security Manager           |
|-------------------------------------------------------------|--------------------|----------------------------|
| Access Security and Risk Management                         | Yes                | Yes                        |
| Access underlying issues (Codacy repositories, Jira issues) | Yes                | Keeps original permissions |

> | Manage integrations                                         | Yes                | ?<!-- TODO confirm  -->    |

Author: TODO: Can a security manager manage integrations?
